### DISCUSSION TITLE
WebDAV怎么设置？

### DISCUSSION THREAD

**Discussion by sunzhuo** [ChatGPTNextWeb/NextChat#4253] [Q&A]
我是这样设置WebDAV的地址，但是设备间好像并没有同步
https://dav.jianguoyun.com/dav/nextchat/nextchat.json

**Comment by VergilWang15** [comment]
webDAV从24年开始的某个版本就一直不可用了，可以尝试用upstash，一些用户反馈是可用的

**Reply by Arthur409** [reply]
upstash现在也不可用了…

**Comment by cacosub7** [comment]
目前坚果云的webDav和upstash在2.11.3中配置好了，检查连通性都没啥问题。但是一点击同步就“同步失败”

**Reply by YANG-Mingcong** [reply]
可能是CORS问题，试试把坚果云的webDAV用proxy方式代理到统一域名下？

**Reply by OliverZhangGit** [reply]
提示失败

**Comment by OliverZhangGit** [comment]
2.15.8 版本，WebDAV同步失败

**Comment by chunfengyao** [comment]
坚果云：`https://dav.jianguoyun.com/dav/`（注意最后的一个斜杠一定要加） 我是检查同步正常而且可以正常同步的。
至于启用代理的那个开关，从devTools里面看到的请求来说：开或不开都是走的代理！

**Reply by OliverZhangGit** [reply]
哪个版本？

**Reply by chunfengyao** [reply]
![image](https://github.com/user-attachments/assets/e392cd5c-2581-4a1c-912e-1196bccd3020)

docker部署的

**Reply by OliverZhangGit** [reply]
依然错误，填的WebDAV地址：https://dav.jianguoyun.com/dav/，另外一个 同步Zotero，相同地址坚果云就可以用

**Reply by chunfengyao** [reply]
浏览器开F12看看呢，返回了什么报错？

**Comment by devonuto** [comment]
Add environment variable "WHITE_WEBDAV_ENDPOINTS", value "https://yourwebdavserver.com/"

Then use that endpoint in the frontend config. This works for me.

EDIT: updated to add S on the end as per comment below

**Reply by chunfengyao** [reply]
https://github.com/ChatGPTNextWeb/NextChat/blob/38093756946ce0d37b24668bbd711f3bed987a7b/app/config/server.ts#L175C1-L176C1

"WHITE_WEBDAV_ENDPOINTS"

**Reply by devonuto** [reply]
That's the one
